internal/services/compute/capacity_reservation_group_resource.go

```diff
@@ -0,0 +1,147 @@
+package compute
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/zones"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceCapacityReservationGroup() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCapacityReservationGroupCreateUpdate,
+		Read:   resourceCapacityReservationGroupRead,
+		Update: resourceCapacityReservationGroupCreateUpdate,
+		Delete: resourceCapacityReservationGroupDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.CapacityReservationGroupID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			// There's a bug in the Azure API where this is returned in upper-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/8068
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
```

Comment:
The resource you copied has bug in `resource_group_name`, it doesn't mean that it has bug in this resource.
```
"resource_group_name": azure.SchemaResourceGroupName(),
```

Reply:
Check this again